pkg/cloudprovider/cloudprovider.go

```diff
@@ -335,3 +351,15 @@ func (c *CloudProvider) instanceToMachine(ctx context.Context, ec2instance *ec2.
 	machine.Status.ProviderID = fmt.Sprintf("aws:///%s/%s", aws.StringValue(ec2instance.Placement.AvailabilityZone), aws.StringValue(ec2instance.InstanceId))
 	return machine
 }
+
+func (c *CloudProvider) getInstanceFromMachine(ctx context.Context, machine *v1alpha5.Machine) (*ec2.Instance, error) {
```

Comment:
I think this logic is repeated in the `Get` function too. Maybe replace this logic there too?